crowbar_framework/lib/upgrade.rb

```diff
@@ -0,0 +1,64 @@
+class Upgrade
+  attr_accessor :current_step, :substep, :current_node, :upgrade_progress
+  attr_reader :steps
+
+  # Return the current state of upgrade process.
+  # We're keeping the information in the file so is accessible by other applications
+  # and different crowbar versions.
+  def initialize
+    @upgrade_progress = {
+      current_step: upgrade_steps_6_7.first,
+      # substep is needed for more complex steps like upgrading the nodes
+      substep: "",
+      # current node is relevant only for the nodes_upgrade step
+      current_node: ""
+    }
+    if progress_file_path.exist?
+      @upgrade_progress = JSON.load(progress_file_path.read)
```

Comment:
There's no point in creating many objects of this. We need to know what's the real status and that is the one saved in the file - e.g. for the cases of some unexpected failure between saving 2 states.